daemon/containerd/image_snapshot.go

```diff
@@ -44,6 +47,39 @@ func (i *ImageService) CreateLayerFromImage(img *image.Image, layerName string,
 	return i.createLayer(descriptor, layerName, rwLayerOpts, nil)
 }
 
+func (i *ImageService) generatePrepareOpts(ctx context.Context, rwLayerOpts *layer.CreateRWLayerOpts) ([]snapshots.Opt, error) {
+	var opts []snapshots.Opt
+
+	if rwLayerOpts != nil && len(rwLayerOpts.StorageOpt) > 0 {
+		for key, val := range rwLayerOpts.StorageOpt {
+			key = strings.ToLower(key)
+			switch key {
+			case "size":
+				size, err := units.RAMInBytes(val)
+				if err != nil {
+					return nil, err
+				}
+
+				if storageDriver := i.StorageDriver(); storageDriver == "windows" {
```

Comment:
Using a `snapshots.Opt` also means that we depend less on the `imageservice.StorageDriver()`, which currently assumes that all containers use the same snapshotter (which currently is the case, but something ideally possible to be specified per-container).